plone/app/querystring/upgrades.py

```diff
@@ -16,3 +16,29 @@ def upgrade_1_to_2_typo_in_registry(context):
     if right_value not in values:
         values.append(right_value)
     registry[name] = values
+
+
+def fix_select_all_existing_collections(context):
+
+    ops_to_fix = [
+        u'portal_type',
+        u'review_state'
+    ]
+    old_val = u"plone.app.querystring.operation.selection.is"
+    new_val = u"plone.app.querystring.operation.selection.any"
+
+    catalog = context.portal_catalog
+    results = catalog.unrestrictedSearchResults(
+        portal_type="Collection"
+    )
+
+    for elem in results:
+        obj = elem.getObject()
+        aux = list()
+        for op in obj.query:
+            if op['i'] in ops_to_fix and op['o'] == old_val:
+                op['o'] = new_val
+            aux.append(op)
+
+        obj.query = aux
```

Comment:
I find the code really hard to read since the variables have rarely a semantic meaning (at least to me).
How about something like this:
```python
indexes_to_fix = [
    u'portal_type',
    u'review_state'
]
old_operator = u"plone.app.querystring.operation.selection.is"
new_operator = u"plone.app.querystring.operation.selection.any"

catalog = context.portal_catalog
brains = catalog.unrestrictedSearchResults(
    portal_type="Collection"
)

for brain in brains:
    obj = brain.getObject()
    fixed_querystring = list()
    for querystring in obj.query:
        if querystring['i'] in indexes_to_fix and querystring['o'] == old_operator:
            querystring['o'] = new_operator
        fixed_query.append(querystring)

    obj.query = fixed_query
    obj.reindexObject()
```